style/spelling-grammar.rst

```diff
@@ -0,0 +1,55 @@
+====================
+Spelling and Grammar
+====================
+
+Some basic principles:
+
+- US English instead of British English
+
+  - Examples:
+  
+    - Write "catalog" (US English). Avoid "catalogue" (British English).
+
+    - Write "license" (US English). Avoid "licence" (British English).
+
+- Compound adjectives should be hypenated whereas compound nouns should not be hypenated
+
+  - Examples:
+
+    - Write "time-series" as an adjective and "time series" as a noun.
+    
+    - Write "real-time" as an adjective (e.g., "real-time analytics") and "real time" as a noun phrase (e.g., "can be queried in real time").
+
+    - Write "open-source" as an adjective and "open source" as a noun.
+
+    - Write "purpose-built" (adjective). Avoid "purpose built".
+
+    - Write "large-scale" (adjective). Avoid "large scale" and "largescale".
+
+    - Write "use case" (noun). Avoid "use-case".
+
+    - Write "shared-nothing" (adjective). Avoid "share-nothing".
+
+    - Write "machine data" (noun). Avoid "machine-data" and "machinedata".
+
+    - Write "data center" (noun). Avoid "data-center" and any variation using "centre" (British English).
+
+  - Exceptions to the rule:
+  
+    - Write "ad hoc" instead of "ad-hoc". (Citation: `StackExchange <https://english.stackexchange.com/a/157332>`_.)
+
+- Drop the hypen from compound words when it is common practice to do so.
+
+  - Examples:
+
+    - Write "database". Avoid "data-base" and "data base".
```

Comment:
Haha I've never seen "data-base" or "data base" before but I guess it can happen !